### PR TITLE
Fix minor navigation UI regressions

### PR DESCRIPTION
**Pull request by kulyk**
This PR fixes a few minor UI regressions caused by new navigation:

**Too wide left border in the query builder**
QB sidebars used to have a `border` style when sidebars are closed, so their borders were always visible on the UI. When we added a new navigation sidebar, these borders became much more noticeable. Fixed by removing border style when QB sidebar is closed.

**Logo icon click area feels too small**
Fixed by making the whole logo container a link.

**Search bar and new button heights**
Followup on #21712 making the search bar and new button have equal 36px height. Had to clean up the `NewButton` component before that (TypeScript, etc.), also `NewButton` is now an HTML `button` instead of `<a />`

### Demo

<details>
<summary>QB Border</summary>

**Before**
qb-border-before

**After**
qb-border-after

</details>

### CODE REVIEW

File: frontend/src/metabase/core/components/Button/Button.tsx [hunk 1]
```diff
@@ -84,11 +87,14 @@ const BaseButton = forwardRef(function BaseButton(
         )}
         {children && (
           <div
```
Comment (ranquild) [inline frontend/src/metabase/core/components/Button/Button.tsx:89]: We could make this a styled component and add to other exported parts, e.g. `Root`:

```
export default Object.assign(Button, {
  Root: ButtonRoot
});
```

The current solution with `contentClassName` makes it very hard to be used with `styled`

Comment (kulyk) [reply frontend/src/metabase/core/components/Button/Button.tsx]: @ranquild done, thanks, it became much better 🚀

**Review by ranquild — APPROVED** [review]

**Review by mazameli — COMMENTED** [review]
Looks good to me, but will this conflict with the tweaks @kdoh and @gusaiani are doing for the sidebar trigger?

**Comment by gusaiani** [comment]
@mazameli If there's any conflict, should be quite easy to fix on the other branch next. 👍

**Pull request merged**